### ISSUE TITLE
Migrate to Visual Studio 2019

### ISSUE THREAD

**Issue by LukeUsher** [Cxbx-Reloaded/Cxbx-Reloaded#1610]
Now that Visual Studio 2019 is released, Microsoft have made it more difficult to acquire Visual Studio 2017, doing so now required a developer account to get access to 'old versions'.

Additionally, Visual Studio 2019 makes it more difficult to acquire the Windows 8.1 SDK, so it won't build out-of-the-box at present.

Based on this, we should look into migrating to Visual Studio 2019.

Since this effects all contributors, we should agree a date for this transition to take place.

There is one caveat: AppVeyor do not currently have a stable build image for VS2019, but this is expected *soon*

See PR #1609 for what is required to migrate (minimal work, but it breaks VS2017 builds)

TLDR: Please discuss potential time-span on VS2019 migration. 

My initial idea would be to transition at the **end of the current month**, so VS2019 will be the required development environment as of **June 1st 2019**, but only if AppVeyor have made VS2019 available by this time.

**Comment by LukeUsher** [comment]
This will be a non-issue after #1611 is merged

**Comment by x1nixmzeng** [comment]
VS2019 preview 2.1 should work on AppVeyor - https://www.appveyor.com/updates/2019/02/11/

It should be trivial to update - although the devenv path is hardcoded in `appveyor-build.bat`

**Comment by RadWolfie** [comment]
Visual Studio 2019 had been released on April 2nd, 2019. Using preview version isn't advice to use. Once they update to the final release (including updates). Although, I do not know if they had done so since the changelog haven't say so.

As for `appveyor-build.bat`, it will no longer exist and use `appveyor.yml` file directly instead. From there, we can update to use 2019. CMake is using hardcoded Visual Studio version than generic one to allow select the newest version of Visual Studio detected on the system.

**Comment by Margen67** [comment]
AppVeyor now supports VS2019.

**Comment by LukeUsher** [comment]
Before we can switch, we also need to make sure that builds made with MSVC2019 are still able to work on Linux with Wine: We don't want to exclude that portion of our userbase.

**Comment by RadWolfie** [comment]
By the way, Visual Studio **16.1** 2019 has CMake 1.14 bundled.

**Comment by RadWolfie** [comment]
One of our tester had reported, couple weeks ago, 2019 version of the build works fine with Linux with Wine.

Currently, Travis CI doesn't have support for VS 2019 yet. However, we are not using Travis CI in this repo. We do welcome a separate pull request if Travis CI does get to the point has VS 2019.


To this conclusion, it is safe to close this ticket.
